napalm/nxos/nxos.py

```diff
@@ -918,10 +942,12 @@ def _load_cfg_from_checkpoint(self) -> None:
 
     def rollback(self) -> None:
         assert isinstance(self.device, NXOSDevice)
-        if self.changed:
-            self.device.rollback(self.rollback_cfg)
-            self._copy_run_start()
-            self.changed = False
+        if not self._check_file_exists(cfg_file=self.rollback_cfg):
+            msg = f"Rollback file '{self.rollback_cfg}' does not exist on device."
+            raise ReplaceConfigException(msg)
```

Comment:
This might not be the right exception i.e. what are the cases where `rollback` gets called. We probably need to look at that a bit more. I generally view a `ReplaceConfigException` as an exception you get when you try to do the replace config operation.

I guess we could raise a generic NapalmException

Reply:
I copied `ReplaceConfigException` from other code that raises that exception during a rollback. I thought about raising `NapalmException` or creating a new `RollbackConfigException`, but I don't have strong preferences here.

Reply:
Okay, let me look some more. I only looked at it quickly (it could make the most sense given what we have previously done).